scripts/git/main_to_google.sh

```diff
@@ -0,0 +1,17 @@
+set -x
+set -e
+set -o pipefail
+
+GREEN_COMMIT="${1:-main}"
+BRANCH="${BRANCH:-main-to-google}"
+
+git sync
+git checkout "${GREEN_COMMIT?}"
+git submodule update
+git checkout -B "${BRANCH?}"
+git push -f fork "${BRANCH?}"
+gh pr create \
+    --base google \
+    --title="Merge main -> google" \
+    --body="$(git log google.. --decorate=no --pretty='format:* %h %<(80,trunc)%s')"
```

Comment:
Hm.. using https://cmder.net/ with sh.exe and cmd.exe, I see

```
λ git log google.. --decorate=no --pretty='format:* %h %<(80,trunc)%s'
The system cannot find the file specified.
```

This works through powershell (ugh..) though:

```
λ powershell.exe "git log google.. --decorate=no --pretty='format:* %h %<(80,trunc)%s'"
* 6706f3d6 Add GitHub action to create PRs merging google -> main (#2482)
...
```

Flipping the quotes around so the inside is `"` works without using powershell:

```suggestion
    --body='$(git log google.. --decorate=no --pretty="format:* %h %<(80,trunc)%s")'
```

Reply:
That doesn't work on linux. The command won't get evaluated. We can keep working at it.

Reply:
Oh, I was just testing the command on its own, outside the script and not as part of a subcommand under `gh pr create`. Both styles of quoting work for me on Linux but only double quotes (`"`) work on Windows outside of powershell. Maybe escape or quote differently in the `--body=`? Or store in a variable then use that variable here? I should also try running the script or full `gh pr create` command, maybe they will work for me.

Reply:
Does cmder use bash? This is a bash script, so I'd expect it to be invoked via bash. I may have been missing a shebang before. Does that help?

Reply:
Well `bash -c "git log main.. --decorate=no --pretty='format:* %h %<(80,trunc)%s'"` works. I can see about switching to bash by default, https://conemu.github.io/en/Tasks.html just has lots of options...

Reply:
Hmmm I feel like if you're trying to run bash scripts in not-bash that's maybe on you ;-P

Reply:
Alright, I actually tried running these main_to_google.sh and it worked:
```
git remote rename origin upstream
set FORK_REMOTE=scotttodd&& bash.exe ..\..\scratch\iree-tmp\main_to_google.sh
```
That created https://github.com/google/iree/pull/2543.

It would still be nice to be able to copy/paste the command snippets involved and run them through other shells, but this is fine for now :D